src/test/java/org/apache/sysds/test/functions/federated/codegen/FederatedCellwiseTmplTest.java

```diff
@@ -147,8 +145,8 @@ private void testFederatedCodegen(ExecMode exec_mode) {
 
 		// generate dataset
 		// matrix handled by two federated workers
-		double[][] X1 = getRandomMatrix(fed_rows, fed_cols, 0, 1, sparsity, 3);
-		double[][] X2 = getRandomMatrix(fed_rows, fed_cols, 0, 1, sparsity, 7);
+		double[][] X1 = getRandomMatrix(fed_rows, fed_cols, 0, 1, 1, 3);
+		double[][] X2 = getRandomMatrix(fed_rows, fed_cols, 0, 1, 1, 7);
```

Comment:
with this you are saying that the test does not care if the input is sparse or dense, if this is the case then to reduce the running time of these test i suggest making all the inputs sparse rather than dense, perhaps set it to 0.1 ?

Reply:
That makes sense for large matrices, but since we also have matrices containing only a few elements I'm afraid of falsifying the test results by changing it. (worst case: all elements of matrix X are 0)
I'm not sure if such a case can happen, can it?